src/broadcast/filelistener/filelistener.cpp

```diff
@@ -0,0 +1,101 @@
+
+#include <QTextCodec>
+#include <QtConcurrentRun>
+
+#include "broadcast/filelistener/filelistener.h"
+#include "broadcast/filelistener/metadatafileworker.h"
+#include "preferences/metadatafilesettings.h"
+
+
+FileListener::FileListener(UserSettingsPointer pConfig)
+        : m_COsettingsChanged(kFileSettingsChanged),
+          m_pConfig(pConfig),
+          m_latestSettings(MetadataFileSettings::getPersistedSettings(pConfig)) {
+
+    MetadataFileWorker *newWorker = new MetadataFileWorker(m_latestSettings.filePath);
```

Comment:
Where is this one deleted?
You can make it a patented_ptr to let Qt take care of this.

Reply:
When the signal finished from the thread is emitted the object is deleted.

Reply:
That is hard to spot. Where exactly is it?
We should follow the rule to pass the pointer with the ownership. Since this is a QObject, the qt object tee may works best as an owner.